dev-util/codeblocks/codeblocks-20.03-r6.ebuild

```diff
@@ -0,0 +1,94 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+WX_GTK_VER="3.0-gtk3"
+
+inherit autotools flag-o-matic wxwidgets xdg
+
+DESCRIPTION="The open source, cross platform, free C, C++ and Fortran IDE"
+HOMEPAGE="https://codeblocks.org/"
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.xz
+	https://dev.gentoo.org/~leio/distfiles/${P}-fortran.tar.xz
+	https://dev.gentoo.org/~leio/distfiles/${P}-fortran-update-v1.7.tar.xz
+	https://dev.gentoo.org/~leio/distfiles/${P}-fortran-update-v1.8.tar.xz
+	https://dev.gentoo.org/~leio/distfiles/${P}-codecompletion-symbolbrowser-update.tar.xz
+"
+
+# USE="fortran" enables FortranProject plugin (updated to v1.8 2021-05-29 [r230])
+# that is delivered with Code::Blocks 20.03 source code.
+# https://sourceforge.net/projects/fortranproject
+# https://cbfortran.sourceforge.io
+
+IUSE="contrib debug fortran"
+
+BDEPEND="virtual/pkgconfig"
+
+RDEPEND="app-arch/zip
+	dev-libs/glib:2
+	>=dev-libs/tinyxml-2.6.2-r3
+	>=dev-util/astyle-3.1-r2:0/3.1
+	x11-libs/gtk+:3
+	x11-libs/wxGTK:${WX_GTK_VER}[X]
+	contrib? (
+		app-admin/gamin
+		app-arch/bzip2
+		app-text/hunspell:=
+		dev-libs/libgamin
+		media-libs/fontconfig
+		sys-libs/zlib
```

Comment:
Still needs to depend on boost. And yeah, I'd perhaps look to use app-alternatives/bzip2 if it works.

Reply:
I didn't think too much about removing `dev-libs/boost:=`, https://github.com/gentoo/gentoo/pull/29186#discussion_r1101480324, on my platform `qa-vdb codeblocks` (`app-portage/iwdevtools-0.12.2`) suggested the same.

Thanks for pushing a commit fixing that ;)

Reply:
You do need the headers at least, but boost is weird and I never really understood how the dep works so I played it safe and placed in RDEPEND.

Reply:
Thanks for your prompt reply.  It seems that there's at least a check during configure.
```
checking for boostlib >=  (102000)... yes
checking whether the Boost::System library is available... yes
checking for exit in -lboost_system... yes
```

Reply:
With `USE=contrib`, `--with-boost-libdir=/usr/lib64` is given to configure.

With `USE=-contrib`, `--without-boost-libdir` is given to configure.

Reply:
Well of course, it's there in the ebuild :)

but seems to work as intended then!